File: historyscan-back/src/test/java/io/sch/historyscan/fake/FileSystemTreeFake.java
package io.sch.historyscan.fake;

import io.sch.historyscan.domain.contexts.analysis.clocrevisions.filesystem.CodeBaseFile;
import io.sch.historyscan.domain.contexts.analysis.clocrevisions.filesystem.FileSystemTree;
import io.sch.historyscan.domain.contexts.analysis.clocrevisions.filesystem.RootFolder;
import org.springframework.core.io.ClassPathResource;

import java.io.IOException;

import static io.sch.historyscan.fake.CodeBaseCommitFake.defaultHistory;

public final class FileSystemTreeFake {

    public static FileSystemTree create(String rootFolderName) throws IOException {
        var codebaseName = "theglobalproject";
        var rootFolder = RootFolder.of(rootFolderName, codebaseName);
        var fsTree = new FileSystemTree(rootFolder);
        var codebaseResource = new ClassPathResource("codebases/theglobalproject");
        var codebasesResource = new ClassPathResource("codebases");
        var rootFile = codebaseResource.getFile();
        fsTree.createFrom(new CodeBaseFile(rootFile, rootFolder,
                codebasesResource.getFile().getPath()));
        return fsTree
                .updateFilesScoreFrom(defaultHistory().commits())
                .then()
                .updateFoldersScore();
    }
}

File: historyscan-back/src/test/kotlin/io/sch/historyscan/fake/FileSystemTreeFake.kt
package io.sch.historyscan.fake

import io.sch.historyscan.domain.contexts.analysis.clocrevisions.filesystem.*
import io.sch.historyscan.domain.contexts.analysis.clocrevisions.filesystem.RootFolder.Companion.of
import org.springframework.core.io.ClassPathResource
import java.io.IOException

object FileSystemTreeFake {
    @Throws(IOException::class)
    fun create(rootFolderName: String?): FileSystemTree? {
        val codebaseName = "theglobalproject"
        val rootFolder = of(rootFolderName!!, codebaseName)
        val fsTree = FileSystemTree(
            rootFolder
        )
        val codebaseResource = ClassPathResource("codebases/theglobalproject")
        val codebasesResource = ClassPathResource("codebases")
        val rootFile = codebaseResource.file
        fsTree.createFrom(
            CodeBaseFile(
                rootFile, rootFolder,
                codebasesResource.file.path
            )
        )
        return fsTree
            .updateFilesScoreFrom(CodeBaseCommitFake.defaultHistory()!!.commits)
            .then()
            .updateFoldersScore()
    }
}
